plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -0,0 +1,521 @@
+    end
+
+    UI:info(_("Building EPUB…"))
+    -- Open the zip file (with .tmp for now, as crengine may still
+    -- have a handle to the final epub_path, and we don't want to
+    -- delete a good one if we fail/cancel later)
+    local epub_path_tmp = epub_path .. ".tmp"
+    local ZipWriter = require("ffi/zipwriter")
+    local epub = ZipWriter:new{}
+    if not epub:open(epub_path_tmp) then
+        logger.dbg("Failed to open epub_path_tmp")
+        return false
+    end
+
+    -- We now create and add all the required epub files
+
+    -- ----------------------------------------------------------------
+    -- /mimetype : always "application/epub+zip"
+    epub:add("mimetype", "application/epub+zip")
+
+    -- ----------------------------------------------------------------
+    -- /META-INF/container.xml : always the same content
+    epub:add("META-INF/container.xml", [[
+<?xml version="1.0"?>
+<container version="1.0" xmlns="urn:oasis:names:tc:opendocument:xmlns:container">
+  <rootfiles>
+    <rootfile full-path="OEBPS/content.opf" media-type="application/oebps-package+xml"/>
+  </rootfiles>
+</container>]])
+    logger.dbg("Added META-INF/container.xml")
+
+    -- ----------------------------------------------------------------
+    -- OEBPS/content.opf : metadata + list of other files (paths relative to OEBPS/ directory)
+    -- Other possible items in this file that are of no interest to crengine :
+    --   In <manifest> :
+    --     <item id="cover" href="title.html" media-type="application/xhtml+xml"/>
+    --     <item id="cover-image" href="images/cover.png" media-type="image/png"/>
+    -- (crengine only uses <meta name="cover" content="cover-image" /> to get the cover image)
+    --   In <spine toc="ncx"> :
+    --     <itemref idref="cover" linear="no"/>
+    --   And a <guide> section :
+    --     <guide>
+    --       <reference href="title.html" type="cover" title="Cover"/>
+    --       <reference href="toc.html" type="toc" title="Table of Contents" href="toc.html" />
+    --     </guide>
+    local content_opf_parts = {}
+    -- head
+    local meta_cover = "<!-- no cover image -->"
+    if include_images and cover_imgid then
+        meta_cover = string.format([[<meta name="cover" content="%s"/>]], cover_imgid)
+    end
+    logger.dbg("meta_cover:", meta_cover)
+    table.insert(content_opf_parts, string.format([[
+<?xml version='1.0' encoding='utf-8'?>
+<package xmlns="http://www.idpf.org/2007/opf"
+        xmlns:dc="http://purl.org/dc/elements/1.1/"
+        unique-identifier="bookid" version="2.0">
+  <metadata>
+    <dc:title>%s</dc:title>
+    <dc:publisher>%s</dc:publisher>
+    %s
+  </metadata>
+  <manifest>
+    <item id="ncx" href="toc.ncx" media-type="application/x-dtbncx+xml"/>
+    <item id="content" href="content.html" media-type="application/xhtml+xml"/>
+    <item id="css" href="stylesheet.css" media-type="text/css"/>
+]], page_htmltitle, Version:getCurrentRevision(), meta_cover))
```

Comment:
Nobody would really see it, but we'll bee be missing the `KOReader` in:
```xml
    <dc:publisher>KOReader v2015.11-1476-g775b7a36</dc:publisher>
``` 
You can just hardcode it in the string `<dc:publisher>KOReader %s</dc:publisher>`

Reply:
Fixed! Thanks.